includes/class-wc-gateway-arkpay.php

```diff
@@ -410,6 +410,12 @@ public function process_payment( $order_id ) {
 
             $transaction = $this->create_arkpay_transaction( $data );
 
+            if ( isset( $transaction->statusCode ) && 400 === $transaction->statusCode ) {
+                if (strpos($transaction->message, 'Currency') !== false && strpos($transaction->message, 'is not allowed') !== false) {
```

Comment:
We don't want to check for 400 error exactly. The sole reason why the currency error has been reported to you is because that was the easiest way to initiate non 200 response. That's how I want this handled, if response is not 200, return the error we get.